### PR TITLE
Kamil analytics

### PR DESCRIPTION
**Pull request by B2o5T**

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -6,5 +6,5 @@
   "access": "restricted",
   "baseBranch": "master",
   "updateInternalDependencies": "patch",
-  "ignore": ["@graphql-codegen/testing", "@graphql-codegen/website", "example-*"]
```
Comment (B2o5T) [inline .changeset/config.json:9]: I renamed `@graphql-codegen/website` to `website` to make consistency between websites, and for `yarn workspace website build`, proposition by @n1ru4l

File: .husky/_/husky.sh [hunk 2]
```diff
@@ -1,31 +0,0 @@
-#!/bin/sh
```
Comment (B2o5T) [inline .husky/_/husky.sh:1]: this file should be ignored

File: package.json [hunk 3]
```diff
@@ -3,8 +3,8 @@
   "private": true,
   "scripts": {
     "postinstall": "patch-package && husky install",
-    "clean": "rimraf node_modules packages/{*,plugins/*/*,presets/*,utils/*}/node_modules",
```
Comment (B2o5T) [inline package.json:6]: a fast glob pattern that results to this
```sh
$ rimraf "{,!(node_modules)/**/}node_modules/"
[
  'node_modules/',
  'examples/programmatic-typescript/node_modules/',
  'packages/graphql-cli-codegen-plugin/node_modules/',
  'packages/graphql-codegen-cli/node_modules/',
  'website/node_modules/',
  'packages/utils/graphql-codegen-testing/node_modules/',
  'packages/utils/plugins-helpers/node_modules/',
  'packages/graphql-codegen-cli/node_modules/make-dir/node_modules/'
]
✨  Done in 0.44s.
```

File: package.json [hunk 4]
```diff
@@ -3,8 +3,8 @@
...
-    "prebuild": "rimraf packages/{*,plugins/*/*,presets/*,utils/*}/dist",
```
Comment (B2o5T) [inline package.json:7]: same, results to this
```sh
$ rimraf "{,!(node_modules)/**/}dist/"
[
  'dist/',
  'packages/graphql-cli-codegen-plugin/dist/',
  'packages/graphql-codegen-cli/dist/',
  'packages/graphql-codegen-core/dist/',
  'packages/presets/gql-tag-operations/dist/',
  'packages/presets/graphql-modules/dist/',
  'packages/presets/import-types/dist/',
  'packages/presets/near-operation-file/dist/',
  'packages/utils/graphql-codegen-testing/dist/',
  'packages/utils/plugins-helpers/dist/',
  'packages/plugins/c-sharp/c-sharp/dist/',
  'packages/plugins/c-sharp/c-sharp-operations/dist/',
  'packages/plugins/flow/flow/dist/',
  'packages/plugins/flow/operations/dist/',
  'packages/plugins/flow/resolvers/dist/',
  'packages/plugins/java/apollo-android/dist/',
  'packages/plugins/java/common/dist/',
  'packages/plugins/java/java/dist/',
  'packages/plugins/java/kotlin/dist/',
  'packages/plugins/java/resolvers/dist/',
  'packages/plugins/other/add/dist/',
  'packages/plugins/other/fragment-matcher/dist/',
  'packages/plugins/other/hasura-allow-list/dist/',
  'packages/plugins/other/introspection/dist/',
  'packages/plugins/other/jsdoc/dist/',
  'packages/plugins/other/schema-ast/dist/',
  'packages/plugins/other/time/dist/',
  'packages/plugins/other/urql-introspection/dist/',
  'packages/plugins/other/visitor-plugin-common/dist/',
  'packages/plugins/typescript/apollo-angular/dist/',
  'packages/plugins/typescript/apollo-client-helpers/dist/',
  'packages/plugins/typescript/document-nodes/dist/',
  'packages/plugins/typescript/enum-array/dist/',
  'packages/plugins/typescript/generic-sdk/dist/',
  'packages/plugins/typescript/gql-tag-operations/dist/',
  'packages/plugins/typescript/graphql-apollo/dist/',
  'packages/plugins/typescript/graphql-files-modules/dist/',
  'packages/plugins/typescript/graphql-request/dist/',
  'packages/plugins/typescript/jit-sdk/dist/',
  'packages/plugins/typescript/mongodb/dist/',
  'packages/plugins/typescript/msw/dist/',
  'packages/plugins/typescript/named-operations-object/dist/',
  'packages/plugins/typescript/oclif/dist/',
  'packages/plugins/typescript/operations/dist/',
  'packages/plugins/typescript/react-apollo/dist/',
  'packages/plugins/typescript/react-apollo-offix/dist/',
  'packages/plugins/typescript/react-query/dist/',
  'packages/plugins/typescript/resolvers/dist/',
  'packages/plugins/typescript/rtk-query/dist/',
  'packages/plugins/typescript/stencil-apollo/dist/',
  'packages/plugins/typescript/type-graphql/dist/',
  'packages/plugins/typescript/typed-document-node/dist/',
  'packages/plugins/typescript/typescript/dist/',
  'packages/plugins/typescript/urql/dist/',
  'packages/plugins/typescript/urql-graphcache/dist/',
  'packages/plugins/typescript/urql-svelte-operations-store/dist/',
  'packages/plugins/typescript/vue-apollo/dist/',
  'packages/plugins/typescript/vue-apollo-smart-ops/dist/',
  'packages/plugins/typescript/vue-urql/dist/'
]
✨  Done in 0.40s
```

File: package.json [hunk 5]
```diff
@@ -120,8 +121,5 @@
     "graphql-language-service-interface": "2.10.2",
     "**/apollo-language-server/graphql": "^16.0.0",
     "**/@types/graphql-upload/graphql": "^16.0.0"
-  },
-  "dependencies": {
-    "patch-package": "6.4.7"
```
Comment (B2o5T) [inline package.json:125]: should be in devdeps

File: packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts [hunk 6]
```diff
@@ -225,7 +225,7 @@ export interface RawResolversConfig extends RawConfig {
    *
    * ## Wrap default types with Partial
    *
-   * You can also specify a custom wrapper for the original type, without overriding the original generated types, use "{T}" to specify the identifier. (for flow, use `$Shape<{T}>`)
```
Comment (B2o5T) [inline packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts:228]: `{T}` must be escaped or to be between \`\`

File: packages/plugins/typescript/generic-sdk/src/config.ts [hunk 7]
```diff
@@ -5,9 +5,9 @@ import { RawClientSideBasePluginConfig } from '@graphql-codegen/visitor-plugin-c
  */
 export interface RawGenericSdkPluginConfig extends RawClientSideBasePluginConfig {
   /**
-   * usingObservableFrom: "import Observable from 'zen-observable';"
+   * usingObservableFrom: `import Observable from 'zen-observable'`
    * OR
-   * usingObservableFrom: "import { Observable } from 'rxjs';"
```
Comment (B2o5T) [inline packages/plugins/typescript/generic-sdk/src/config.ts:10]: same for `{ Observable }`

File: packages/plugins/typescript/resolvers/src/config.ts [hunk 8]
```diff
@@ -66,7 +66,7 @@ export interface TypeScriptResolversPluginConfig extends RawResolversConfig {
    */
   customResolveInfo?: string;
   /**
-   * @description You can provide your custom ResolveFn instead the default. It has to be a type that uses the generics <TResult, TParent, TContext, TArgs>
```
Comment (B2o5T) [inline packages/plugins/typescript/resolvers/src/config.ts:69]: `<` should be escaped or to be between \`\`

File: website/docs-templates/c-sharp-operations.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 id: c-sharp-operations
 ---
 
-The `c-sharp-operations` plugin generates C# methods for the resolvers signature.
```
Comment (B2o5T) [inline website/docs-templates/c-sharp-operations.md:5]: more correct

File: package.json [hunk 10]
```diff
@@ -3,8 +3,8 @@
   "private": true,
   "scripts": {
     "postinstall": "patch-package && husky install",
-    "clean": "rimraf node_modules packages/{*,plugins/*/*,presets/*,utils/*}/node_modules",
-    "prebuild": "rimraf packages/{*,plugins/*/*,presets/*,utils/*}/dist",
+    "clean": "rimraf \"{,!(node_modules)/**/}node_modules/\"",
```
Comment (B2o5T) [inline package.json:6]: a fast glob pattern that results to this
```sh
$ rimraf "{,!(node_modules)/**/}node_modules/"
[
  'node_modules/',
  'examples/programmatic-typescript/node_modules/',
  'packages/graphql-cli-codegen-plugin/node_modules/',
  'packages/graphql-codegen-cli/node_modules/',
  'website/node_modules/',
  'packages/utils/graphql-codegen-testing/node_modules/',
  'packages/utils/plugins-helpers/node_modules/',
  'packages/graphql-codegen-cli/node_modules/make-dir/node_modules/'
]
✨  Done in 0.44s.
```

File: package.json [hunk 11]
```diff
@@ -3,8 +3,8 @@
...
+    "prebuild": "rimraf \"{,!(node_modules)/**/}dist/\"",
```
Comment (B2o5T) [inline package.json:7]: same, results to this
```sh
$ rimraf "{,!(node_modules)/**/}dist/"
[
  'dist/',
  'packages/graphql-cli-codegen-plugin/dist/',
  'packages/graphql-codegen-cli/dist/',
  'packages/graphql-codegen-core/dist/',
  'packages/presets/gql-tag-operations/dist/',
  'packages/presets/graphql-modules/dist/',
  'packages/presets/import-types/dist/',
  'packages/presets/near-operation-file/dist/',
  'packages/utils/graphql-codegen-testing/dist/',
  'packages/utils/plugins-helpers/dist/',
  'packages/plugins/c-sharp/c-sharp/dist/',
  'packages/plugins/c-sharp/c-sharp-operations/dist/',
  'packages/plugins/flow/flow/dist/',
  'packages/plugins/flow/operations/dist/',
  'packages/plugins/flow/resolvers/dist/',
  'packages/plugins/java/apollo-android/dist/',
  'packages/plugins/java/common/dist/',
  'packages/plugins/java/java/dist/',
  'packages/plugins/java/kotlin/dist/',
  'packages/plugins/java/resolvers/dist/',
  'packages/plugins/other/add/dist/',
  'packages/plugins/other/fragment-matcher/dist/',
  'packages/plugins/other/hasura-allow-list/dist/',
  'packages/plugins/other/introspection/dist/',
  'packages/plugins/other/jsdoc/dist/',
  'packages/plugins/other/schema-ast/dist/',
  'packages/plugins/other/time/dist/',
  'packages/plugins/other/urql-introspection/dist/',
  'packages/plugins/other/visitor-plugin-common/dist/',
  'packages/plugins/typescript/apollo-angular/dist/',
  'packages/plugins/typescript/apollo-client-helpers/dist/',
  'packages/plugins/typescript/document-nodes/dist/',
  'packages/plugins/typescript/enum-array/dist/',
  'packages/plugins/typescript/generic-sdk/dist/',
  'packages/plugins/typescript/gql-tag-operations/dist/',
  'packages/plugins/typescript/graphql-apollo/dist/',
  'packages/plugins/typescript/graphql-files-modules/dist/',
  'packages/plugins/typescript/graphql-request/dist/',
  'packages/plugins/typescript/jit-sdk/dist/',
  'packages/plugins/typescript/mongodb/dist/',
  'packages/plugins/typescript/msw/dist/',
  'packages/plugins/typescript/named-operations-object/dist/',
  'packages/plugins/typescript/oclif/dist/',
  'packages/plugins/typescript/operations/dist/',
  'packages/plugins/typescript/react-apollo/dist/',
  'packages/plugins/typescript/react-apollo-offix/dist/',
  'packages/plugins/typescript/react-query/dist/',
  'packages/plugins/typescript/resolvers/dist/',
  'packages/plugins/typescript/rtk-query/dist/',
  'packages/plugins/typescript/stencil-apollo/dist/',
  'packages/plugins/typescript/type-graphql/dist/',
  'packages/plugins/typescript/typed-document-node/dist/',
  'packages/plugins/typescript/typescript/dist/',
  'packages/plugins/typescript/urql/dist/',
  'packages/plugins/typescript/urql-graphcache/dist/',
  'packages/plugins/typescript/urql-svelte-operations-store/dist/',
  'packages/plugins/typescript/vue-apollo/dist/',
  'packages/plugins/typescript/vue-apollo-smart-ops/dist/',
  'packages/plugins/typescript/vue-urql/dist/'
]
✨  Done in 0.40s
```

File: packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts [hunk 12]
```diff
@@ -225,7 +225,7 @@ export interface RawResolversConfig extends RawConfig {
...
+   * You can also specify a custom wrapper for the original type, without overriding the original generated types, use `{T}` to specify the identifier. (for flow, use `$Shape<{T}>`)
```
Comment (B2o5T) [inline packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts:228]: `{T}` must be escaped or to be between \`\`

File: packages/plugins/typescript/generic-sdk/src/config.ts [hunk 13]
```diff
@@ -5,9 +5,9 @@ import { RawClientSideBasePluginConfig } from '@graphql-codegen/visitor-plugin-c
...
+   * usingObservableFrom: `import { Observable } from 'rxjs'`
```
Comment (B2o5T) [inline packages/plugins/typescript/generic-sdk/src/config.ts:10]: same for `{ Observable }`

File: packages/plugins/typescript/resolvers/src/config.ts [hunk 14]
```diff
@@ -66,7 +66,7 @@ export interface TypeScriptResolversPluginConfig extends RawResolversConfig {
...
+   * @description You can provide your custom ResolveFn instead the default. It has to be a type that uses the generics `<TResult, TParent, TContext, TArgs>`
```
Comment (B2o5T) [inline packages/plugins/typescript/resolvers/src/config.ts:69]: `<` should be escaped or to be between \`\`

File: website/docs-templates/c-sharp-operations.md [hunk 15]
```diff
@@ -2,7 +2,7 @@
...
+The `c-sharp-operations` plugin generates C# methods for the resolvers' signature.
```
Comment (B2o5T) [inline website/docs-templates/c-sharp-operations.md:5]: more correct

**Review by charlypoly — APPROVED** [review]
⚡

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**